include/fiction/algorithms/simulation/sidb/operational_domain.hpp

```diff
@@ -50,108 +138,78 @@ namespace fiction
 };
-
+/**
+ * This function compares two parameter points, `p1` and `p2`, for equality with tolerance
+ * (`std::numeric_limits<double>::epsilon()`). It checks if the difference between each coordinate of the parameter
+ * points is within a specified tolerance.
+ *
+ * @param p1 The first parameter point to compare.
+ * @param p2 The second parameter point to compare.
+ * @return `true` if the parameter points are considered equal within the specified tolerance, `false otherwise.
+ */
+static bool check_parameter_points_for_equality(const parameter_point& p1, const parameter_point& p2)
+{
+    constexpr auto tolerance = physical_constants::POP_STABILITY_ERR;
+    return std::abs(p1.x - p2.x) < tolerance && std::abs(p1.y - p2.y) < tolerance;
+}
+/**
+ * This function searches for a parameter point, specified by the `key`, in the provided map
+ * `map` with tolerance. It compares each key in the map to the specified key using the
+ * `check_parameter_points_for_equality` function.
+ *
+ * @tparam MapType The type of the map containing parameter points as keys.
+ * @param map The map containing parameter points as keys and associated values.
+ * @param key The parameter point to search for in the map.
+ * @return An iterator to the found parameter point in the map, or `map.cend()` if not found.
+ */
+template <typename MapType>
+[[maybe_unused]] static typename MapType::const_iterator
+find_parameter_point_with_tolerance(const MapType& map, const typename MapType::key_type& key)
+{
+    static_assert(std::is_same_v<typename MapType::key_type, parameter_point>, "Map key type must be parameter_point");
+    const auto compare_keys = [&key](const auto& pair) { return check_parameter_points_for_equality(pair.first, key); };
+    return std::find_if(map.cbegin(), map.cend(), compare_keys);
+}
+/**
+ * This function searches for a parameter point, specified by the `key`, in the provided map
+ * `map` with tolerance. It compares each key in the map to the specified key using the
+ * `check_parameter_points_for_equality` function.
+ *
+ * @tparam MapType The type of the map containing parameter points as keys.
+ * @param map The map containing parameter points as keys and associated values.
+ * @param key The parameter point to search for in the map.
+ * @return An iterator to the found parameter point in the map, or `map.end()` if not found.
+ */
+template <typename MapType>
+[[maybe_unused]] static typename MapType::const_iterator find_key_with_tolerance(const MapType&                    map,
+                                                                                 const typename MapType::key_type& key)
+{
+    static_assert(std::is_arithmetic_v<typename MapType::key_type>, "Map key type must be numeric");
+    constexpr double tolerance = fiction::physical_constants::POP_STABILITY_ERR;
+    auto compare_keys = [&key, &tolerance](const auto& pair) { return std::abs(pair.first - key) < tolerance; };
```

Comment:
You only need this if `MapType::key_type` is a floating-point type, right?

Reply:
@marcelwa Yes, should we then use ``static_assert(std::is_floating_point_v<typename MapType::key_type>, "Map key type must be numeric");``?

Reply:
"floating-point" instead of "numeric". 

Either that (but it restricts the entire usability to floating-point values exclusively) or you use an `if constexpr` to conditionally trigger if the keys are floating-point